src/microbe_stage/organelle_components/SlimeJetComponent.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using Godot;
+
+/// <summary>
+///   Slime-powered jet for adding bursts of speed
+/// </summary>
+public class SlimeJetComponent : ExternallyPositionedComponent
+{
+    private bool active;
+
+    private AnimationPlayer? animation;
+
+    private Compound mucilage = null!;
+
+    /// <summary>
+    ///   Whether this jet is currently secreting slime and animating
+    /// </summary>
+    public bool Active
+    {
+        get => active;
+        set
+        {
+            active = value;
+
+            if (animation != null)
+            {
+                // Play the animation if active, and vice versa
+                animation.PlaybackSpeed = active ? 1.0f : 0.0f;
+            }
+        }
+    }
+
+    public override void UpdateAsync(float delta)
+    {
+        // Visual positioning code
+        base.UpdateAsync(delta);
+
+        // Movement force
+        var microbe = organelle!.ParentMicrobe!;
+
+        if (microbe.PhagocytosisStep != PhagocytosisPhase.None)
+            return;
+
+        var movement = CalculateMovementForce(microbe, delta);
+
+        if (movement != Vector3.Zero)
+            microbe.AddMovementForce(movement);
+    }
+
+    /// <summary>
+    ///   Determines the movement impulse imparted by this jet by ejecting some mucilage
+    /// </summary>
+    public Vector3 CalculateMovementForce(Microbe microbe, float delta)
+    {
+        if (!Active)
+            return Vector3.Zero;
+
+        var currentCellRotation = microbe.GlobalTransform.basis.Quat();
+        var direction = GetDirection();
+
+        // Eject mucilage at the maximum rate in a direction opposite to this organelle's rotation
+        var slimeSecreted = microbe.EjectCompound(
```

Comment:
This is unsafe code to trigger from `UpdateAsync`
See: https://github.com/Revolutionary-Games/Thrive/issues/3270 (absorb and eject compound operations can't be multithreaded).

Reply:
Is that specific to the cloud system or does that problem include adding/subtracting compounds from microbes? If it's both, I don't know how to fix this, since I need to know the amount of mucilage we can eject to calculate the movement force here.

Reply:
Adding and subtracting compounds is 100% safe and even intended for the async operation. That is for the *current* microbe, no other microbe's state may be modified. The way I've gotten around this in the microbe class is to have a "temporary" variable hold a result the async method calculated and then applying it in the sync part, though in your case it might be easier to have the Microbe class emit the slime and each slime component just adds some queued slime to be emitted.

Reply:
Ok, that's what I expected. I ended up implementing the temporary variable approach anyway, so take a quick look at that to check it's what you had in mind.

Reply:
Seems fine to me.